### ISSUE TITLE
Deferred loading for field from field.gradient()

### ISSUE THREAD

**Issue by JoostSH** [OceanParcels/parcels#371]
Hi,
I am calculating the vorticity field from a 2D velocity fieldset using the built in ```Field.gradient()``` method.
It works fine when the fieldset is fully loaded with
```python
fieldset = FieldSet.from_netcdf(filenames, variables, dimensions, full_load=True)
fieldVORT = Field('VORT', fieldset.V.gradient()[0] - fieldset.U.gradient()[1], grid=fieldset.U.grid)
fieldset.add_field(fieldVORT)
```
However, when the deferred loading is used (```full_load=False```), I get the following error log:

```
---------------------------------------------------------------------------
AxisError                                 Traceback (most recent call last)
<ipython-input-16-9da17ee2622e> in <module>()
----> 1 fieldVORT = Field('VORT',fieldset.V.gradient()[0]-fieldset.U.gradient()[1],grid=fieldset.U.grid)
      2 fieldset.add_field(fieldVORT)

/home/jhemmen/miniconda3/envs/py2_parcels/lib/python2.7/site-packages/parcels/field.pyc in gradient(self)
    449         if not self.grid.cell_edge_sizes:
    450             self.calc_cell_edge_sizes()
--> 451         dFdy = np.gradient(self.data, axis=-2) / self.grid.cell_edge_sizes['y']
    452         dFdx = np.gradient(self.data, axis=-1) / self.grid.cell_edge_sizes['x']
    453         return dFdx, dFdy

/home/jhemmen/miniconda3/envs/py2_parcels/lib/python2.7/site-packages/numpy/lib/function_base.pyc in gradient(f, *varargs, **kwargs)
   1689         axes = tuple(range(N))
   1690     else:
-> 1691         axes = _nx.normalize_axis_tuple(axes, N)
   1692 
   1693     len_axes = len(axes)

/home/jhemmen/miniconda3/envs/py2_parcels/lib/python2.7/site-packages/numpy/core/numeric.pyc in normalize_axis_tuple(axis, ndim, argname, allow_duplicate)
   1504     except TypeError:
   1505         axis = tuple(axis)
-> 1506     axis = tuple(normalize_axis_index(ax, ndim, argname) for ax in axis)
   1507     if not allow_duplicate and len(set(axis)) != len(axis):
   1508         if argname:

/home/jhemmen/miniconda3/envs/py2_parcels/lib/python2.7/site-packages/numpy/core/numeric.pyc in <genexpr>((ax,))
   1504     except TypeError:
   1505         axis = tuple(axis)
-> 1506     axis = tuple(normalize_axis_index(ax, ndim, argname) for ax in axis)
   1507     if not allow_duplicate and len(set(axis)) != len(axis):
   1508         if argname:

AxisError: axis -2 is out of bounds for array of dimension 0
```
My question would be: Is it possible to also implement the deferred loading to Fields that are created from the original fieldset?

**Comment by erikvansebille** [comment]
Thanks for reporting the bug, @JoostSH! Sorry that we haven't responded for a while. This shouldn't be too hard to fix though, so hopefully we'll fix this soon(ish)

**Comment by erikvansebille** [comment]
See PR #383 for a (hopefully) working fix to this issue. @JoostSH, could you test it and confirm that it works for your case too?

**Comment by JoostSH** [comment]
Hi @erikvansebille! thanks for the fix.
Creating deferred loading fields using the ```Field.gradient()``` method is working with PR #383.

I would like to create and sample fields of Vorticity, Strain and Okubo-Weiss Parameter and therefore need to subtract or add the field gradients. It seems ike this is not working currently and would need a new method for the ```Field``` class (e.g. ```Field.addition()```). Or is there a way to do that?

Of course, sampling the gradients and calculating the parameters named-above in a custom sampling kernel should work fine and provide a solution.

**Comment by erikvansebille** [comment]
Thanks for the quick response, @JoostSH. Good to know that the gradient calculation works for you!

As for manipulation of Fields, it's true that we don't have very good API for that yet. The issue was also raised in #326. Of course, you can always directly manipulate the `Field.data`, which are the numpy arrays holding the actual data. But we might want to provide some additional methods like overloading the `+` and `-` operators for `Fields`, like we have also done for `Kernels` as a way to concatenate them. I'll open a separate Issue on this idea

**Comment by erikvansebille** [comment]
Issue is fixed in #383
